source/templates.tex

```diff
@@ -6178,7 +6178,12 @@
 function\iref{special}, the program is ill-formed.
 
 \pnum
-The \grammarterm{declaration} in an \grammarterm{explicit-instantiation} and the \grammarterm{declaration} produced by the corresponding substitution into the templated function, variable, or class are two declarations of the same entity.
+The \grammarterm{declaration} in an \grammarterm{explicit-instantiation} and
+the \grammarterm{declaration} produced by the corresponding substitution
+into the templated function, variable, or class
+are two declarations of the same entity.
+An \grammarterm{explicit-instantiation} of a function template
+shall not specify a contract condition\iref{dcl.attr.contract}.
```

Comment:
This is not an appropriate place for this rule (it's placed between a normative rule and its note and example). In fact, we often place such rules with the rules about the prohibited thing, not in the explicit instantiation rules. For example, [dcl.stc] disallows storage classes in explicit instantiations, and [except.spec] gives the rules for noexcept in explicit instantiations. But I would note that some other related rules are in [temp.explicit]p1: "An explicit instantiation of a function template, member function of a class template, or variable template shall not use the inline or constexpr specifiers." (And I'm not sure where we disallow an explicit instantiation from having default arguments...)

But this is also redundant. [dcl.attr.grammar]p5 says "No attribute-specifier-seq shall appertain to an explicit instantiation". So I think we should either drop this change or convert it to a note in p1.

Reply:
Removed due to redundancy. I'm not adding a note; we should decide whether we want the "explicit instantiation" restriction in [temp.explicit] or where the other feature is defined, and consistently put notes in the "other" places. See #2188, for a later time.